### PR TITLE
Refactored logging.ColoredFormatter to avoid deepcopy.

### LINKED ISSUE #7528: Logger does deepcopy of record which fails in some environments
**Software Versions**
* Python: 3.7
* OS: Win
* Kivy: 2.1.0.dev0

**Describe the bug**
Doing `logging.exception(e)` from within a particular script fails when there's an object in the stack that can't be copied. The following the is the error I get:

```
 --- Logging error ---
 Traceback (most recent call last):
   File "G:/Python/libs/ceed/ceed/analysis/merge_data.py", line 1665, in <module>
     align = alignment[experiment] = merger.get_alignment()
   File "G:/Python/libs/ceed/ceed/analysis/merge_data.py", line 1378, in get_alignment
     return self._get_alignment(True)
   File "G:/Python/libs/ceed/ceed/analysis/merge_data.py", line 1394, in _get_alignment
     'Cannot find any experiment in the MCS parsed data')
 AlignmentException: Cannot find any experiment in the MCS parsed data
 
 During handling of the above exception, another exception occurred:
 
 Traceback (most recent call last):
   File "C:\Users\Matthew Einhorn\AppData\Local\Programs\Python\Python37\Lib\logging\__init__.py", line 1034, in emit
     msg = self.format(record)
   File "C:\Users\Matthew Einhorn\AppData\Local\Programs\Python\Python37\Lib\logging\__init__.py", line 880, in format
     return fmt.format(record)
   File "g:\python\libs\kivy\kivy\logger.py", line 291, in format
     record = copy.deepcopy(record)
   File "E:\Python\Python37\lib\copy.py", line 180, in deepcopy
     y = _reconstruct(x, memo, *rv)
   File "E:\Python\Python37\lib\copy.py", line 280, in _reconstruct
     state = deepcopy(state, memo)
   File "E:\Python\Python37\lib\copy.py", line 150, in deepcopy
     y = copier(x, memo)
   File "E:\Python\Python37\lib\copy.py", line 240, in _deepcopy_dict
     y[deepcopy(key, memo)] = deepcopy(value, memo)
   File "E:\Python\Python37\lib\copy.py", line 150, in deepcopy
     y = copier(x, memo)
   File "E:\Python\Python37\lib\copy.py", line 220, in _deepcopy_tuple
     y = [deepcopy(a, memo) for a in x]
   File "E:\Python\Python37\lib\copy.py", line 220, in <listcomp>
     y = [deepcopy(a, memo) for a in x]
   File "E:\Python\Python37\lib\copy.py", line 169, in deepcopy
     rv = reductor(4)
 TypeError: can't pickle traceback objects
 Call stack:
   File "G:/Python/libs/ceed/ceed/analysis/merge_data.py", line 1673, in <module>
     logging.exception(str(e))
 Message: 'Cannot find any experiment in the MCS parsed data'
 Arguments: ()

Process finished with exit code 0
```

It seems to be because of this line in kivy/logger.py: `record = copy.deepcopy(record)`.

### PR DESCRIPTION
**Pull request by Julian-O**
This is a clean up of kivy.logger code, as described in #7959 It has no effect to the way log files appear or how they are used. It is an internal refactor.

It has an incidental effect of no longer using `deepcopy()` on logRecords, so it fixes #7528 and #7585.

Maintainer merge checklist
* [x] Title is descriptive/clear for inclusion in release notes.
* [x] Applied a `Component: xxx` label.
* [ ] Applied the `api-deprecation` or `api-break` label.
* [ ] Applied the `release-highlight` label to be highlighted in release notes.
* [x] Added to the milestone version it was merged into.
* [x] **Unittests** are included in PR.
* [ ] Properly documented, including `versionadded`, `versionchanged` as needed.

### CODE REVIEW

**Review by misl6 — CHANGES_REQUESTED** [review]
Hi @Julian-O !
LGTM. Thank you!

I've just some minor things that may be addressed before the merge:

- `copy` is still imported even if it's not used anymore
- `randint` is still imported even if it's not used anymore (not your fault)

Also (and could be fixed later), I've seen that there's still a `try/except` for Python 2 compat, feel free to remove it, as we don't support Python 2 anymore.

**Comment by Julian-O** [comment]
I agree with everything you said - and I have already fixed them all, and more, in the *next* PR I have planned.

Can we please merge this one now if I promise to give you a PR with those changes in the next couple of days? I am trying to balance giving you small, incremental improvements that are self-contained and easy-to-review versus how long it will take to get through the backlog (on the order of 7 more PRs?) with review delays between each one. (Happy to hear advice on this.)

**Comment by misl6** [comment]
> I am trying to balance giving you small, incremental improvements that are self-contained and easy-to-review
I like it 💯 

Yeah, the only one which is really related is `copy`, but if you promise to remove it in a while (and you promised it 😄 ), there's no issue to get it merged now.

Thank you again!

**Review by misl6 — APPROVED** [review]

**Pull request merged**
